include/fixed_containers/fixed_map_raw_view.hpp

```diff
@@ -0,0 +1,186 @@
+#pragma once
+
+#include "fixed_containers/fixed_red_black_tree_nodes.hpp"
+#include "fixed_containers/fixed_red_black_tree_types.hpp"
+#include "fixed_containers/fixed_red_black_tree_view.hpp"
+#include "fixed_containers/map_entry_raw_view.hpp"
+
+#include <cstddef>
+#include <iterator>
+
+namespace fixed_containers
+{
+
+class FixedMapRawView
+{
+    using Compactness = fixed_red_black_tree_detail::RedBlackTreeNodeColorCompactness;
+    using StorageType = fixed_red_black_tree_detail::RedBlackTreeStorageType;
+    using NodeIndex = fixed_red_black_tree_detail::NodeIndex;
+
+public:
+    class Iterator
+    {
+    private:
+
+        // I use Entry so we can match UnorderedMapRawView's API where,
+        // *it returns an object with .key() and .value() methods.
+        // You could also do this with MapEntry but it's a lot clunkier.
```

Comment:
Did you mean using `MapEntryRawView` would be clunkier? It seems like using `MapEntryRawView` wouldn't be bad to use here, unless I'm missing something

Reply:
Sure you could have a MapEntryRawView as a member variable, but then you would have to reinitialize this variable every you updated the base_iterator_, which just seems clunky. Also, we'd store redundant data.